File: 11455 Behold my quadrangle/solution.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return *(int*)a > *(int*)b;
}

int main() {
    int sides[4], cases;
    scanf("%d", &cases);
    while(cases--) {
        for(int i = 0; i < 4; ++i)
			scanf("%d", &sides[i]);
		qsort(sides, 4, sizeof(int), compare);
		if(sides[0] == sides[1] && sides[1] == sides[2] && sides[2] == sides[3])
			puts("square");
		else if(sides[0] == sides[1] && sides[2] == sides[3])
			puts("rectangle");
		else if(sides[0] + sides[1] + sides[2] > sides[3])
			puts("quadrangle");
		else
			puts("banana");
    }
    return 0;
}

File: 11455 Behold my quadrangle/solution.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
        int[] sides = new int[4];
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        st.nextToken();
        int cases = (int)st.nval;
        StringBuilder output = new StringBuilder();
        while(cases-- > 0) {
            for(int i = 0; i < 4; ++i) {
                st.nextToken();
    			sides[i] = (int)st.nval;
            }
    		Arrays.sort(sides);
    		if(sides[0] == sides[1] && sides[1] == sides[2] && sides[2] == sides[3])
    			output.append("square\n");
    		else if(sides[0] == sides[1] && sides[2] == sides[3])
    			output.append("rectangle\n");
    		else if(sides[0] + sides[1] + sides[2] > sides[3])
    			output.append("quadrangle\n");
    		else
    			output.append("banana\n");
        }
        System.out.print(output);
	}
}
